File: src/node/jvm/api/text/Characters.ts
import { CharactersInstance } from "./CharactersInstance";

export const Characters: CharactersInstance = {
    isDigit(c: string): boolean {
        if (c.length !== 1) return false;
        const code = c.charCodeAt(0);
        return code >= 48 && code <= 57;  // '0'..'9'
    },

    isLetter(c: string): boolean {
        if (c.length !== 1) return false;
        const code = c.charCodeAt(0);
        return (code >= 65 && code <= 90)    // 'A'..'Z'
            || (code >= 97 && code <= 122);  // 'a'..'z'
    }
};

File: src/java/jvm/api/text/Characters.java
package jvm.api.text;

import magma.annotate.Actual;
import magma.annotate.Namespace;

@Namespace
public final class Characters {
    @Actual
    public static boolean isDigit(final char c) {
        return Character.isDigit(c);
    }

    @Actual
    public static boolean isLetter(final char c) {
        return Character.isLetter(c);
    }
}
